cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -270,51 +269,44 @@ func GetSchedulerProbeAddress(cfg *kubeadmapi.ClusterConfiguration) string {
 	return "127.0.0.1"
 }
 
-// GetEtcdProbeAddress returns the etcd probe address
-func GetEtcdProbeAddress(cfg *kubeadmapi.Etcd) string {
+// GetEtcdProbeURL takes a kubeadm Etcd configuration object and attempts to parse
+// the first URL in the listen-metrics-urls argument, returning an etcd probe hostname,
+// port and scheme
+func GetEtcdProbeURL(cfg *kubeadmapi.Etcd) (string, int, v1.URIScheme) {
+	localhost := "127.0.0.1"
 	if cfg.Local != nil && cfg.Local.ExtraArgs != nil {
-		if arg, exists := cfg.Local.ExtraArgs[etcdListenClientURLsArg]; exists {
-			// Use the first url in the listen-client-urls if multiple url's are specified.
+		if arg, exists := cfg.Local.ExtraArgs[EtcdListenMetricsURLsArg]; exists {
+			// Use the first url in the listen-metrics-urls if multiple URL's are specified.
 			if strings.ContainsAny(arg, ",") {
 				arg = strings.Split(arg, ",")[0]
 			}
 			parsedURL, err := url.Parse(arg)
-			if err != nil || parsedURL.Hostname() == "" {
-				return "127.0.0.1"
+			if err != nil {
+				return localhost, kubeadmconstants.EtcdMetricsPort, v1.URISchemeHTTP
 			}
-			// Return the IP if the URL contains an address instead of a name.
-			if ip := net.ParseIP(parsedURL.Hostname()); ip != nil {
-				// etcdctl doesn't support auto-converting zero addresses into loopback addresses
-				if ip.Equal(net.IPv4zero) {
-					return "127.0.0.1"
-				}
-				if ip.Equal(net.IPv6zero) {
-					return net.IPv6loopback.String()
-				}
-				return ip.String()
+			// Parse scheme
+			scheme := v1.URISchemeHTTP
+			if parsedURL.Scheme == "https" {
+				scheme = v1.URISchemeHTTPS
```

Comment:
Does HTTPS work well without providing client certificate? 

When  `--listen-metrics-urls=https://127.0.0.1:2381`:
```sh
# curl https://127.0.0.1:2381/health
curl: (60) Peer's Certificate issuer is not recognized.
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.

# curl -k https://127.0.0.1:2381/health
curl: (58) NSS: client certificate not found (nickname not specified)
```
When  `--listen-metrics-urls=http://127.0.0.1:2381`:
```sh
# curl http://127.0.0.1:2381/health
{"health":"true"}
```
Maybe we should only support HTTP Scheme now?